python/dgl/graphbolt/impl/csc_sampling_graph.py

```diff
@@ -108,6 +110,59 @@ def total_num_edges(self) -> int:
         """
         return self._c_csc_graph.num_edges()
 
+    @property
+    def num_nodes(self) -> Union[int, Dict[str, int]]:
+        """The number of nodes in the graph.
+        - If the graph is homogenous, returns an integer.
+        - If the graph is heterogenous, returns a dictionary.
+
+        Returns
+        -------
+        Union[int, Dict[str, int]]
+            The number of nodes. Integer indicates the total nodes number of a
+            homogenous graph; dict indicates nodes number per node types of a
+            heterogenous graph.
+
+        TODO[Mingbang]: Add code examples to demonstrate the usage of num_nodes
+        """
+        offset = self.node_type_offset
+
+        # Homogenous.
+        if offset is None or self.metadata is None:
+            return self._c_csc_graph.num_nodes()
+
+        else:
+            # First, the graph is treated as heterogenous, then the return
+            # type depends on the length of the dictionary.
+            num_nodes_per_type = {
+                _type: offset[_idx + 1] - offset[_idx]
+                for _type, _idx in self.metadata.node_type_to_id.items()
+            }
+
+            if len(num_nodes_per_type) > 1:  # True Heterogenous.
+                return num_nodes_per_type
+            else:  # Homogenous.
+                return self._c_csc_graph.num_nodes()
+
+    @property
+    def num_edges(self) -> Union[int, Dict[str, int]]:
+        """[NOT BEEN IMPLEMENTED YET]
+
+        TODO[Mingbang]: Discuss the implementation of this API.
+
+        The number of edges in the graph.
+        - If the graph is homogenous, returns an integer.
+        - If the graph is heterogenous, returns a dictionary.
+
+        Returns
+        -------
+        Union[int, Dict[str, int]]
+            The number of edges. Integer indicates the total edges number of a
+            homogenous graph; dict indicates edges number per edge types of a
+            heterogenous graph.
+        """
+        raise self.total_num_edges
```

Comment:
Just remove this should be fine. Let the parent class throw an not implemented error is better than silently provide a compromised implementation.

Reply:
```
python/dgl/graphbolt/impl/csc_sampling_graph.py:78:0: W0223: Method 'num_edges' is abstract in class 'SamplingGraph' but is not overridden (abstract-method)
```

message from https://dgl-ci-result.s3.us-west-2.amazonaws.com/dgl/PR-6395/6/6/logs/logs_dir/tmped_v7uvf.log

maybe I should remove this api from the parent class?